rules/codelingo/effective-go/single-method-interface-name/codelingo.yaml

```diff
@@ -12,17 +12,31 @@ tenets:
         title: Single Method Interface Name
         body: |
           By convention, one-method interfaces are named by the method name plus an -er suffix 
-          or similar modification to construct an agent noun: Reader, Writer, Formatter, CloseNotifier etc.
-
-          There are a number of such names and it's productive to honor them and the function names they capture. 
-          Read, Write, Close, Flush, String and so on have canonical signatures and meanings. To avoid confusion, 
+          Read, Write, Close, Flush, String and so on have canonical signatures and meanings
+          and it's productive to honor them and the function names they capture. To avoid confusion, 
           don't give your method one of those names unless it has the same signature and meaning. Conversely, 
           if your type implements a method with the same meaning as a method on a well-known type, give it the 
           same name and signature; call your string-converter method String not ToString.
+
+          ## Bad code
+          ```golang
+          // ReadByte contains a basic Read method which populates the given byte slice with data. 
+          type ReadByte interface {
+            Read() []byte
+          } 
+          ```
+
+          ## Good code
+          ```golang
+          // Reader contains a basic Read method which populates the given byte slice with data.
+          type Reader interface {
+            Read() []byte
+          }
+          ```
```

Comment:
@meadows115 I think this code example might confuse the reader, because the description above says to avoid using the standard method names, such as `Read`. Can you come up with another example? Could be anything like `Download` + `Downloader`, perhaps with a bad code name like `WebClient`.